builtin/credential/approle/path_login.go

```diff
@@ -33,7 +33,11 @@ func pathLogin(b *backend) *framework.Path {
 // Returns the Auth object indicating the authentication and authorization information
 // if the credentials provided are validated by the backend.
 func (b *backend) pathLoginUpdate(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	role, roleName, metadata, err := b.validateCredentials(req, data)
+	if req.Connection == nil || req.Connection.RemoteAddr == "" {
+		return nil, fmt.Errorf("failed to get connection information")
```

Comment:
Oh the reason for this being here are 2 things: 1) This is an internal error. If the connection information is not passed to the backend, then there is something wrong, even if it is not hitting the use case that would need this value. 2) Currently we have the req object shoved down into validateCredentials. But if we want to fetch the source IP *after* the validation, the request object needs to be sent further down to the bound validation methods. A string instead looked much simpler.

Let me know if you insist on passing the `req` object and I'll do it. Either way is fine for me.